packages/as-types/assembly/amount.ts

```diff
@@ -28,62 +32,89 @@ export class Amount {
   /**
    * Adds two amounts and return results in a new one.
    *
-   * @param  a - Amount to add.
+   * @param  other - The other Amount to add.
+   *
+   * @returns
+   * - A Result containing the new Amount.
+   * - An error message if the currency is not the same between the two amounts.
+   * - An error message if the addition would overflow 'u64.MAX_VALUE'.
```

Comment:
```suggestion
   * @returns  
   * A Result object:
   * - containing the new Amount object if the operation is successful,
   * - containing an error message if
   *    - the currency of this Amount object and the other Amount object are not the same, or
   *    - the addition would cause an overflow of the 'u64.MAX_VALUE' limit.
```